### PR TITLE
Create dockerfile for doxygen

### PR DESCRIPTION
**Pull request by richardbeare**
Extensions to generate package docs for python/java/R

### CODE REVIEW

File: Utilities/GenerateDocs/docker/cmd.sh [hunk 1]
```diff
@@ -20,4 +22,16 @@ cmake \
 ninja SimpleITK-doc
 
 cd SimpleITK-build/Documentation/
+
 tar -zcvf ${OUT_DIR}/simpleitk_doxygen_html_${SIMPLEITK_GIT_TAG}.tar.gz html
+rm ${here}/SimpleITK/Wrapping/R/Packaging/SimpleITK/man/*.Rd
+
+cd ${here}/SimpleITK/Utilities/GenerateDocs/
+cp /work/bin/config_vars.sh .
+./SwigDocUpdate.sh
```
Comment (blowekamp) [inline Utilities/GenerateDocs/docker/cmd.sh:42]: This is a tricky one... This script is coming from the checkout repo to build SimpleITK... How can we test edits to the script?

**Review by blowekamp — COMMENTED** [review]
Please create a new file say "cmd_swig_docs.sh" which contains these new steps. Keep the updates in the Dockerfile, but update "run.sh" to run docker twice. The first time with just cmd.sh to generate the Doxygen, and the second with "cmd_swig_docs.sh" to create the Swig/Rd documentation.

If you add the script after the docker tag on the command line it will be passes as an argument to the entry point which should run the script as the proper user. Something like this:
```
docker run --rm -t .. -v $(pwd):/work/io  ${DOCKER_TAG} cmd_swig_docs.sh
```

**Comment by richardbeare** [comment]
I was testing changes via the docker bash entry point. Is documenting that approach acceptable?

**Comment by blowekamp** [comment]
Sounds good!

On Jan 11, 2019, at 5:10 PM, Richard Beare <notifications@github.com<mailto:notifications@github.com>> wrote:


I was testing changes via the docker bash entry point. Is documenting that approach acceptable?

**Comment by richardbeare** [comment]
Query - is it possible to retain the results of cmd.sh in the docker container before running cmd_swig_docs.sh? It seems to me I'd need to rerun the build in the second script?

**Comment by blowekamp** [comment]
What do you need in the second script? Just the produced doxygen and a couple scripts, right?

**Comment by richardbeare** [comment]
Yep, the scripts are there because they get copied into work/bin/

However the checked out copy and build of SimpleITK, which includes the
necessary xml files, which were in /tmp, disappear next time we run the
container.

**Comment by blowekamp** [comment]
Ahh. In cmd.sh can we tar.gz the xml filed too?

**Comment by richardbeare** [comment]
That's all I could think of too - but is there any advantage in having two
scripts? Going that direction will lead to another build scenario that is
different to the original style.

**Comment by blowekamp** [comment]
I suppose I am just viewing it from a "Makefile" perspective. Step 1: takes A as input and produces B; Step 2: takes B as input and produces C.

I suppose it doesn't make much of a difference. Lets get something which works easily, and is documented in!

**Comment by richardbeare** [comment]
OK,
I'll get onto it.

**Comment by richardbeare** [comment]
Ready for testing - more documentation, a single cmd script, but some additional variables supported to make testing changes to doc scripts a bit easier.

**Comment by blowekamp** [comment]
Works for me! Thanks!

**Pull request merged**